code/modules/station_goals/bsa.dm

```diff
@@ -304,8 +304,14 @@ DEFINE_BUFFER_HANDLER(/obj/machinery/bsa/middle)
 	var/list/options = gps_locators
 	if(area_aim)
 		options += GLOB.teleportlocs
-	var/V = input(user,"Select target", "Select target",null) in options|null
-	target = options[V]
+	var/victim = tgui_input_list(user, "Select target", "Artillery Targeting", options)
+	if(isnull(victim))
+		return
+	if(isnull(options[victim]))
+		return
+	target = options[victim]
+	var/datum/component/gps/log_target = target
+	log_game("[key_name(user)] has aimed the artillery strike at [get_area_name(log_target.parent)].")
```

Comment:
```suggestion
	log_game("[key_name(user)] has aimed the bluespace artillery (BSA) strike at [AREACOORD(log_target.parent)].")
```

Reply:
get_area_name is intentional here. At this point in time it's only gotten the area and has yet to select a tile. I'll commit the readability improvement though